99.wc_object/wc.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require_relative 'wc_file'
+
+class Wc
+  def initialize(argv)
+    @filenames = argv
+    @options = {}
+    extract_options
```

Comment:
メソッドを呼び出すとその中でインスタンス変数の値が変更される、というのは予期せぬ副作用なので設計としてあまりよろしくないです。

以下のような呼び出しができた方がベターです。

```suggestion
    @options, @filenames = extract_options(argv)
```

参考 https://blog.jnito.com/entry/2023/07/12/115733

Reply:
修正しました。

ここなのですが、extract_options(オプションの抜粋する)メソッドの中で、
オプションだけでなく、ファイル名の取得もしています。責務が二つになってしまっていますが、あまり良くないでしょうか？(双方とも、OptionParserオブジェクトを利用するので、現状一ヶ所に書いてしまっています。)

Reply:
good questionですね！良い観点です。
たしかに責務は2つになってしまいますが、OptionParserを使う以上やむを得ない、ということで許容してしまっていいと思います。